### ISSUE TITLE
Instance launch using --from-json fails with CreateVnicDetails

### ISSUE THREAD

**Issue by craigcarl-oracle** [oracle/oci-cli#152]
When passing options to `oci compute instance launch` using `--from-json` the launch will fail if the subnetId is supplied via createVnicDetails. The launch will succeed if subnetId is supplied via LaunchInstanceDetails. 

```
$ oci --version
2.5.11

$ oci compute instance launch --region us-ashburn-1 --from-json file://launch_fail.json
ServiceError:
{
    "code": "InvalidParameter",
    "message": "Invalid subnet: null",
    "opc-request-id": "06E626243D3B4311A7435E9D920ECDC2/5990B996726FF7DB82EC12AF589A5275/50E86A40D97E7622A58EE345735F7215",
    "status": 400
}
```
launch_fail.json -
```
{
	"compartmentId": "ocid1.tenancy.oc1..aaaaaaaa7wdgkzk6am4qxann4rclbaurqy7wlbcbowyqchmewjbzdc2uc6fq",
	"availabilityDomain": "Uocm:US-ASHBURN-AD-1",
	"definedTags": {},
	"freeformTags": {},
	"sourceDetails": {
		"sourceType": "image",
		"imageId": "ocid1.image.oc1.iad.aaaaaaaa4bfsnhv2cd766tiw5oraw2as7g27upxzvu7ynqwipnqfcfwqskla"
	},
	"shape": "VM.Standard2.1",
	"displayName": "instance-20190524-1356",
	"metadata": {
		"ssh_authorized_keys": "ssh-rsa AAAAB3NzaC1..."
	},
	"createVnicDetails": {
		"subnetId": "ocid1.subnet.oc1.iad.aaaaaaaapgbjk6c3ptpripb2xctmuhx2ykmecxj6juncenyeeiff73h5ryka"
	},
	"agentConfig": {
		"isMonitoringDisabled": false
	}
}
```

```
$ oci compute instance launch --region us-ashburn-1 --from-json file://launch.json
{...
"lifecycle-state": "PROVISIONING"
...}
```
launch.json -
```
{
	"compartmentId": "ocid1.tenancy.oc1..aaaaaaaa7wdgkzk6am4qxann4rclbaurqy7wlbcbowyqchmewjbzdc2uc6fq",
	"availabilityDomain": "Uocm:US-ASHBURN-AD-1",
	"subnetId": "ocid1.subnet.oc1.iad.aaaaaaaapgbjk6c3ptpripb2xctmuhx2ykmecxj6juncenyeeiff73h5ryka",
	"definedTags": {},
	"freeformTags": {},
	"sourceDetails": {
		"sourceType": "image",
		"imageId": "ocid1.image.oc1.iad.aaaaaaaa4bfsnhv2cd766tiw5oraw2as7g27upxzvu7ynqwipnqfcfwqskla"
	},
	"shape": "VM.Standard2.1",
	"displayName": "instance-20190524-1356",
	"metadata": {
		"ssh_authorized_keys": "ssh-rsa AAAAB3NzaC1..."
	},
	"agentConfig": {
		"isMonitoringDisabled": false
	}
}
```

**Comment by paul-hummel-oracle** [comment]
Confirmed the bug and was able to reproduce.

Also tried the same with the API directly with curl --
https://docs.cloud.oracle.com/iaas/api/#/en/iaas/20160918/Instance/LaunchInstance
https://docs.cloud.oracle.com/iaas/Content/API/Concepts/signingrequests.htm#Bash
The API worked ok, so the problem is isolated to the CLI.

**Comment by srikanthkumbham** [comment]
@craigcarl-oracle this issue has been fixed in the latest release (`2.5.13`)
